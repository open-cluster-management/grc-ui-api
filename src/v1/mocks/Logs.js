/** *****************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2018. All Rights Reserved.
 *
 * Note to U.S. Government Users Restricted Rights:
 * Use, duplication or disclosure restricted by GSA ADP Schedule
 * Contract with IBM Corp.
 ****************************************************************************** */

export const mockClusterResponse = {
  body: {
    items: [
      {
        kind: 'Cluster',
        apiVersion: 'clusterregistry.k8s.io/v1alpha1',
        metadata: {
          name: 'managed-cluster',
          namespace: 'default',
          selfLink: '/apis/clusterregistry.k8s.io/v1alpha1/namespaces/default/clusters/cluster1',
          uid: 'd9f3a5aa-9f19-11e8-855e-f2b998610544',
          resourceVersion: '136558',
          creationTimestamp: '2018-08-13T16:56:41Z',
          labels: {
            cloud: 'IBM',
            datacenter: 'toronto',
            environment: 'Dev',
            name: 'managed-cluster',
            owner: 'marketing',
            region: 'US',
            vendor: 'ICP',
          },
          annotations: {
            'mcm.ibm.com/user-group': 'c3lzdGVtOnNlcnZpY2VhY2NvdW50cyxzeXN0ZW06c2VydmljZWFjY291bnRzOmt1YmUtc3lzdGVtLHN5c3RlbTphdXRoZW50aWNhdGVk',
            'mcm.ibm.com/user-identity': 'c3lzdGVtOnNlcnZpY2VhY2NvdW50Omt1YmUtc3lzdGVtOmRlZmF1bHQ=',
          },
          finalizers: [
            'finalizer.hcm.ibm.com',
          ],
        },
        spec: {
          kubernetesApiEndpoints: {
            serverEndpoints: [
              {
                serverAddress: '9.42.80.212:8001',
              },
            ],
          },
          authInfo: {},
        },
        status: {
          conditions: [
            {
              type: 'OK',
              status: '',
              lastHeartbeatTime: '2018-08-15T19:41:20Z',
              lastTransitionTime: null,
            },
          ],
        },
      },
    ],
  },
};

export const mockLogsResponse = {
  body: 'here is the logs',
};
