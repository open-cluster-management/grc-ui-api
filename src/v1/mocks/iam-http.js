/** *****************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2018, 2019. All Rights Reserved.
 *
 * Note to U.S. Government Users Restricted Rights:
 * Use, duplication or disclosure restricted by GSA ADP Schedule
 * Contract with IBM Corp.
 ****************************************************************************** */

export default function createMockHttp() {
  const state = {
    namespaces: {
      items: [
        { // non cluster namespace
          metadata: {
            name: 'default',
          },
        },
        { // non cluster namespace
          metadata: {
            name: 'kube-system',
          },
        },
        { // non cluster namespace
          metadata: {
            name: 'policy-namespace',
          },
        },
        { // cluster namespace
          metadata: {
            name: 'cluster1',
          },
        },
        { // cluster namespace
          metadata: {
            name: 'local-cluster',
          },
        },
      ],
    },
  };

  return async function MockLib(params) {
    switch (true) {
      case params.url.includes('resourceType=namespace'):
      default:
        return state.namespaces;
    }
  };
}
