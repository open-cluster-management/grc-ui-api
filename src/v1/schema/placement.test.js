/** *****************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2019. All Rights Reserved.
 * Copyright (c) 2020 Red Hat, Inc.
 *
 * Note to U.S. Government Users Restricted Rights:
 * Use, duplication or disclosure restricted by GSA ADP Schedule
 * Contract with IBM Corp.
 ****************************************************************************** */

import supertest from 'supertest';
import nock from 'nock';
import server, { GRAPHQL_PATH } from '../index';
import { mockPlacementListResponse, hostUrl } from '../mocks/PlacementList';

describe('Placement Resolver', () => {
  beforeAll(() => {
    // specify the url to be intercepted
    const APIServer = nock(hostUrl);

    APIServer.get('/mcm.ibm.com/v1alpha1/namespaces/default/placementpolicies')
      .reply(200, mockPlacementListResponse);
    APIServer.get('/mcm.ibm.com/v1alpha1/namespaces/kube-system/placementpolicies')
      .reply(200, mockPlacementListResponse);
    APIServer.get('/mcm.ibm.com/v1alpha1/namespaces/mcm/placementpolicies')
      .reply(200, mockPlacementListResponse);
    APIServer.get('/mcm.ibm.com/v1alpha1/namespaces/cluster1/placementpolicies')
      .reply(200, mockPlacementListResponse);
    APIServer.get('/mcm.ibm.com/v1alpha1/namespaces/clusterhub/placementpolicies')
      .reply(200, mockPlacementListResponse);
  });

  test('Correctly Resolves Placement Policy List Query', (done) => {
    supertest(server)
      .post(GRAPHQL_PATH)
      .send({
        query: `
        {
          placementPolicies {
            metadata {
              annotations
              name
              namespace
              creationTimestamp
              selfLink
            }
            clusterLabels
            clusterReplicas
            resourceSelector
            status
            raw
          }
        }
      `,
      })
      .end((err, res) => {
        expect(JSON.parse(res.text)).toMatchSnapshot();
        done();
      });
  });
});
