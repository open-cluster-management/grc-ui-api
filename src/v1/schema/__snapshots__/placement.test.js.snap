// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Placement Resolver Correctly Resolves Placement Policy List Query 1`] = `
Object {
  "data": Object {
    "placementPolicies": Array [
      Object {
        "clusterLabels": Object {
          "matchLabels": Object {
            "cloud": "IBM",
          },
        },
        "clusterReplicas": null,
        "metadata": Object {
          "annotations": Object {
            "mcm.ibm.com/user-group": "c3lzdGVtOnNlcnZpY2VhY2NvdW50cyxzeXN0ZW06c2VydmljZWFjY291bnRzOmt1YmUtc3lzdGVtLHN5c3RlbTphdXRoZW50aWNhdGVk",
            "mcm.ibm.com/user-identity": "c3lzdGVtOnNlcnZpY2VhY2NvdW50Omt1YmUtc3lzdGVtOmRlZmF1bHQ=",
          },
          "creationTimestamp": "2019-07-29T17:41:59Z",
          "name": "placement-my-policy",
          "namespace": "mcm",
          "selfLink": "/apis/apps.open-cluster-management.io/v1/namespaces/mcm/placementrules/placement-my-policy",
        },
        "raw": Object {
          "apiVersion": "apps.open-cluster-management.io/v1",
          "kind": "PlacementRule",
          "metadata": Object {
            "annotations": Object {
              "mcm.ibm.com/user-group": "c3lzdGVtOnNlcnZpY2VhY2NvdW50cyxzeXN0ZW06c2VydmljZWFjY291bnRzOmt1YmUtc3lzdGVtLHN5c3RlbTphdXRoZW50aWNhdGVk",
              "mcm.ibm.com/user-identity": "c3lzdGVtOnNlcnZpY2VhY2NvdW50Omt1YmUtc3lzdGVtOmRlZmF1bHQ=",
            },
            "creationTimestamp": "2019-07-29T17:41:59Z",
            "name": "placement-my-policy",
            "namespace": "mcm",
            "resourceVersion": "60",
            "selfLink": "/apis/apps.open-cluster-management.io/v1/namespaces/mcm/placementrules/placement-my-policy",
            "uid": "2a52a098-b228-11e9-926f-eae85572ef1a",
          },
          "spec": Object {
            "clusterSelector": Object {
              "matchLabels": Object {
                "cloud": "IBM",
              },
            },
            "resourceHint": Object {},
            "resourceSelector": Object {},
          },
          "status": Object {
            "decisions": Array [
              Object {
                "clusterName": "clusterhub",
                "clusterNamespace": "clusterhub",
              },
              Object {
                "clusterName": "cluster1",
                "clusterNamespace": "cluster1",
              },
            ],
          },
        },
        "resourceSelector": Object {},
        "status": Object {
          "decisions": Array [
            Object {
              "clusterName": "clusterhub",
              "clusterNamespace": "clusterhub",
            },
            Object {
              "clusterName": "cluster1",
              "clusterNamespace": "cluster1",
            },
          ],
        },
      },
      Object {
        "clusterLabels": Object {
          "matchLabels": Object {
            "cloud": "IBM",
          },
        },
        "clusterReplicas": null,
        "metadata": Object {
          "annotations": Object {
            "mcm.ibm.com/user-group": "c3lzdGVtOnNlcnZpY2VhY2NvdW50cyxzeXN0ZW06c2VydmljZWFjY291bnRzOmt1YmUtc3lzdGVtLHN5c3RlbTphdXRoZW50aWNhdGVk",
            "mcm.ibm.com/user-identity": "c3lzdGVtOnNlcnZpY2VhY2NvdW50Omt1YmUtc3lzdGVtOmRlZmF1bHQ=",
          },
          "creationTimestamp": "2019-07-29T17:45:39Z",
          "name": "placement-policy-ma",
          "namespace": "mcm",
          "selfLink": "/apis/apps.open-cluster-management.io/v1/namespaces/mcm/placementrules/placement-policy-ma",
        },
        "raw": Object {
          "apiVersion": "apps.open-cluster-management.io/v1",
          "kind": "PlacementRule",
          "metadata": Object {
            "annotations": Object {
              "mcm.ibm.com/user-group": "c3lzdGVtOnNlcnZpY2VhY2NvdW50cyxzeXN0ZW06c2VydmljZWFjY291bnRzOmt1YmUtc3lzdGVtLHN5c3RlbTphdXRoZW50aWNhdGVk",
              "mcm.ibm.com/user-identity": "c3lzdGVtOnNlcnZpY2VhY2NvdW50Omt1YmUtc3lzdGVtOmRlZmF1bHQ=",
            },
            "creationTimestamp": "2019-07-29T17:45:39Z",
            "name": "placement-policy-ma",
            "namespace": "mcm",
            "resourceVersion": "98",
            "selfLink": "/apis/apps.open-cluster-management.io/v1/namespaces/mcm/placementrules/placement-policy-ma",
            "uid": "ada3957f-b228-11e9-926f-eae85572ef1a",
          },
          "spec": Object {
            "clusterSelector": Object {
              "matchLabels": Object {
                "cloud": "IBM",
              },
            },
            "resourceHint": Object {},
            "resourceSelector": Object {},
          },
          "status": Object {
            "decisions": Array [
              Object {
                "clusterName": "clusterhub",
                "clusterNamespace": "clusterhub",
              },
              Object {
                "clusterName": "cluster1",
                "clusterNamespace": "cluster1",
              },
            ],
          },
        },
        "resourceSelector": Object {},
        "status": Object {
          "decisions": Array [
            Object {
              "clusterName": "clusterhub",
              "clusterNamespace": "clusterhub",
            },
            Object {
              "clusterName": "cluster1",
              "clusterNamespace": "cluster1",
            },
          ],
        },
      },
      Object {
        "clusterLabels": Object {
          "matchLabels": Object {
            "cloud": "IBM",
          },
        },
        "clusterReplicas": null,
        "metadata": Object {
          "annotations": Object {
            "mcm.ibm.com/user-group": "c3lzdGVtOnNlcnZpY2VhY2NvdW50cyxzeXN0ZW06c2VydmljZWFjY291bnRzOmt1YmUtc3lzdGVtLHN5c3RlbTphdXRoZW50aWNhdGVk",
            "mcm.ibm.com/user-identity": "c3lzdGVtOnNlcnZpY2VhY2NvdW50Omt1YmUtc3lzdGVtOmRlZmF1bHQ=",
          },
          "creationTimestamp": "2019-07-29T17:41:59Z",
          "name": "placement-my-policy",
          "namespace": "mcm",
          "selfLink": "/apis/apps.open-cluster-management.io/v1/namespaces/mcm/placementrules/placement-my-policy",
        },
        "raw": Object {
          "apiVersion": "apps.open-cluster-management.io/v1",
          "kind": "PlacementRule",
          "metadata": Object {
            "annotations": Object {
              "mcm.ibm.com/user-group": "c3lzdGVtOnNlcnZpY2VhY2NvdW50cyxzeXN0ZW06c2VydmljZWFjY291bnRzOmt1YmUtc3lzdGVtLHN5c3RlbTphdXRoZW50aWNhdGVk",
              "mcm.ibm.com/user-identity": "c3lzdGVtOnNlcnZpY2VhY2NvdW50Omt1YmUtc3lzdGVtOmRlZmF1bHQ=",
            },
            "creationTimestamp": "2019-07-29T17:41:59Z",
            "name": "placement-my-policy",
            "namespace": "mcm",
            "resourceVersion": "60",
            "selfLink": "/apis/apps.open-cluster-management.io/v1/namespaces/mcm/placementrules/placement-my-policy",
            "uid": "2a52a098-b228-11e9-926f-eae85572ef1a",
          },
          "spec": Object {
            "clusterSelector": Object {
              "matchLabels": Object {
                "cloud": "IBM",
              },
            },
            "resourceHint": Object {},
            "resourceSelector": Object {},
          },
          "status": Object {
            "decisions": Array [
              Object {
                "clusterName": "clusterhub",
                "clusterNamespace": "clusterhub",
              },
              Object {
                "clusterName": "cluster1",
                "clusterNamespace": "cluster1",
              },
            ],
          },
        },
        "resourceSelector": Object {},
        "status": Object {
          "decisions": Array [
            Object {
              "clusterName": "clusterhub",
              "clusterNamespace": "clusterhub",
            },
            Object {
              "clusterName": "cluster1",
              "clusterNamespace": "cluster1",
            },
          ],
        },
      },
      Object {
        "clusterLabels": Object {
          "matchLabels": Object {
            "cloud": "IBM",
          },
        },
        "clusterReplicas": null,
        "metadata": Object {
          "annotations": Object {
            "mcm.ibm.com/user-group": "c3lzdGVtOnNlcnZpY2VhY2NvdW50cyxzeXN0ZW06c2VydmljZWFjY291bnRzOmt1YmUtc3lzdGVtLHN5c3RlbTphdXRoZW50aWNhdGVk",
            "mcm.ibm.com/user-identity": "c3lzdGVtOnNlcnZpY2VhY2NvdW50Omt1YmUtc3lzdGVtOmRlZmF1bHQ=",
          },
          "creationTimestamp": "2019-07-29T17:45:39Z",
          "name": "placement-policy-ma",
          "namespace": "mcm",
          "selfLink": "/apis/apps.open-cluster-management.io/v1/namespaces/mcm/placementrules/placement-policy-ma",
        },
        "raw": Object {
          "apiVersion": "apps.open-cluster-management.io/v1",
          "kind": "PlacementRule",
          "metadata": Object {
            "annotations": Object {
              "mcm.ibm.com/user-group": "c3lzdGVtOnNlcnZpY2VhY2NvdW50cyxzeXN0ZW06c2VydmljZWFjY291bnRzOmt1YmUtc3lzdGVtLHN5c3RlbTphdXRoZW50aWNhdGVk",
              "mcm.ibm.com/user-identity": "c3lzdGVtOnNlcnZpY2VhY2NvdW50Omt1YmUtc3lzdGVtOmRlZmF1bHQ=",
            },
            "creationTimestamp": "2019-07-29T17:45:39Z",
            "name": "placement-policy-ma",
            "namespace": "mcm",
            "resourceVersion": "98",
            "selfLink": "/apis/apps.open-cluster-management.io/v1/namespaces/mcm/placementrules/placement-policy-ma",
            "uid": "ada3957f-b228-11e9-926f-eae85572ef1a",
          },
          "spec": Object {
            "clusterSelector": Object {
              "matchLabels": Object {
                "cloud": "IBM",
              },
            },
            "resourceHint": Object {},
            "resourceSelector": Object {},
          },
          "status": Object {
            "decisions": Array [
              Object {
                "clusterName": "clusterhub",
                "clusterNamespace": "clusterhub",
              },
              Object {
                "clusterName": "cluster1",
                "clusterNamespace": "cluster1",
              },
            ],
          },
        },
        "resourceSelector": Object {},
        "status": Object {
          "decisions": Array [
            Object {
              "clusterName": "clusterhub",
              "clusterNamespace": "clusterhub",
            },
            Object {
              "clusterName": "cluster1",
              "clusterNamespace": "cluster1",
            },
          ],
        },
      },
      Object {
        "clusterLabels": Object {
          "matchLabels": Object {
            "cloud": "IBM",
          },
        },
        "clusterReplicas": null,
        "metadata": Object {
          "annotations": Object {
            "mcm.ibm.com/user-group": "c3lzdGVtOnNlcnZpY2VhY2NvdW50cyxzeXN0ZW06c2VydmljZWFjY291bnRzOmt1YmUtc3lzdGVtLHN5c3RlbTphdXRoZW50aWNhdGVk",
            "mcm.ibm.com/user-identity": "c3lzdGVtOnNlcnZpY2VhY2NvdW50Omt1YmUtc3lzdGVtOmRlZmF1bHQ=",
          },
          "creationTimestamp": "2019-07-29T17:41:59Z",
          "name": "placement-my-policy",
          "namespace": "mcm",
          "selfLink": "/apis/apps.open-cluster-management.io/v1/namespaces/mcm/placementrules/placement-my-policy",
        },
        "raw": Object {
          "apiVersion": "apps.open-cluster-management.io/v1",
          "kind": "PlacementRule",
          "metadata": Object {
            "annotations": Object {
              "mcm.ibm.com/user-group": "c3lzdGVtOnNlcnZpY2VhY2NvdW50cyxzeXN0ZW06c2VydmljZWFjY291bnRzOmt1YmUtc3lzdGVtLHN5c3RlbTphdXRoZW50aWNhdGVk",
              "mcm.ibm.com/user-identity": "c3lzdGVtOnNlcnZpY2VhY2NvdW50Omt1YmUtc3lzdGVtOmRlZmF1bHQ=",
            },
            "creationTimestamp": "2019-07-29T17:41:59Z",
            "name": "placement-my-policy",
            "namespace": "mcm",
            "resourceVersion": "60",
            "selfLink": "/apis/apps.open-cluster-management.io/v1/namespaces/mcm/placementrules/placement-my-policy",
            "uid": "2a52a098-b228-11e9-926f-eae85572ef1a",
          },
          "spec": Object {
            "clusterSelector": Object {
              "matchLabels": Object {
                "cloud": "IBM",
              },
            },
            "resourceHint": Object {},
            "resourceSelector": Object {},
          },
          "status": Object {
            "decisions": Array [
              Object {
                "clusterName": "clusterhub",
                "clusterNamespace": "clusterhub",
              },
              Object {
                "clusterName": "cluster1",
                "clusterNamespace": "cluster1",
              },
            ],
          },
        },
        "resourceSelector": Object {},
        "status": Object {
          "decisions": Array [
            Object {
              "clusterName": "clusterhub",
              "clusterNamespace": "clusterhub",
            },
            Object {
              "clusterName": "cluster1",
              "clusterNamespace": "cluster1",
            },
          ],
        },
      },
      Object {
        "clusterLabels": Object {
          "matchLabels": Object {
            "cloud": "IBM",
          },
        },
        "clusterReplicas": null,
        "metadata": Object {
          "annotations": Object {
            "mcm.ibm.com/user-group": "c3lzdGVtOnNlcnZpY2VhY2NvdW50cyxzeXN0ZW06c2VydmljZWFjY291bnRzOmt1YmUtc3lzdGVtLHN5c3RlbTphdXRoZW50aWNhdGVk",
            "mcm.ibm.com/user-identity": "c3lzdGVtOnNlcnZpY2VhY2NvdW50Omt1YmUtc3lzdGVtOmRlZmF1bHQ=",
          },
          "creationTimestamp": "2019-07-29T17:45:39Z",
          "name": "placement-policy-ma",
          "namespace": "mcm",
          "selfLink": "/apis/apps.open-cluster-management.io/v1/namespaces/mcm/placementrules/placement-policy-ma",
        },
        "raw": Object {
          "apiVersion": "apps.open-cluster-management.io/v1",
          "kind": "PlacementRule",
          "metadata": Object {
            "annotations": Object {
              "mcm.ibm.com/user-group": "c3lzdGVtOnNlcnZpY2VhY2NvdW50cyxzeXN0ZW06c2VydmljZWFjY291bnRzOmt1YmUtc3lzdGVtLHN5c3RlbTphdXRoZW50aWNhdGVk",
              "mcm.ibm.com/user-identity": "c3lzdGVtOnNlcnZpY2VhY2NvdW50Omt1YmUtc3lzdGVtOmRlZmF1bHQ=",
            },
            "creationTimestamp": "2019-07-29T17:45:39Z",
            "name": "placement-policy-ma",
            "namespace": "mcm",
            "resourceVersion": "98",
            "selfLink": "/apis/apps.open-cluster-management.io/v1/namespaces/mcm/placementrules/placement-policy-ma",
            "uid": "ada3957f-b228-11e9-926f-eae85572ef1a",
          },
          "spec": Object {
            "clusterSelector": Object {
              "matchLabels": Object {
                "cloud": "IBM",
              },
            },
            "resourceHint": Object {},
            "resourceSelector": Object {},
          },
          "status": Object {
            "decisions": Array [
              Object {
                "clusterName": "clusterhub",
                "clusterNamespace": "clusterhub",
              },
              Object {
                "clusterName": "cluster1",
                "clusterNamespace": "cluster1",
              },
            ],
          },
        },
        "resourceSelector": Object {},
        "status": Object {
          "decisions": Array [
            Object {
              "clusterName": "clusterhub",
              "clusterNamespace": "clusterhub",
            },
            Object {
              "clusterName": "cluster1",
              "clusterNamespace": "cluster1",
            },
          ],
        },
      },
      Object {
        "clusterLabels": Object {
          "matchLabels": Object {
            "cloud": "IBM",
          },
        },
        "clusterReplicas": null,
        "metadata": Object {
          "annotations": Object {
            "mcm.ibm.com/user-group": "c3lzdGVtOnNlcnZpY2VhY2NvdW50cyxzeXN0ZW06c2VydmljZWFjY291bnRzOmt1YmUtc3lzdGVtLHN5c3RlbTphdXRoZW50aWNhdGVk",
            "mcm.ibm.com/user-identity": "c3lzdGVtOnNlcnZpY2VhY2NvdW50Omt1YmUtc3lzdGVtOmRlZmF1bHQ=",
          },
          "creationTimestamp": "2019-07-29T17:41:59Z",
          "name": "placement-my-policy",
          "namespace": "mcm",
          "selfLink": "/apis/apps.open-cluster-management.io/v1/namespaces/mcm/placementrules/placement-my-policy",
        },
        "raw": Object {
          "apiVersion": "apps.open-cluster-management.io/v1",
          "kind": "PlacementRule",
          "metadata": Object {
            "annotations": Object {
              "mcm.ibm.com/user-group": "c3lzdGVtOnNlcnZpY2VhY2NvdW50cyxzeXN0ZW06c2VydmljZWFjY291bnRzOmt1YmUtc3lzdGVtLHN5c3RlbTphdXRoZW50aWNhdGVk",
              "mcm.ibm.com/user-identity": "c3lzdGVtOnNlcnZpY2VhY2NvdW50Omt1YmUtc3lzdGVtOmRlZmF1bHQ=",
            },
            "creationTimestamp": "2019-07-29T17:41:59Z",
            "name": "placement-my-policy",
            "namespace": "mcm",
            "resourceVersion": "60",
            "selfLink": "/apis/apps.open-cluster-management.io/v1/namespaces/mcm/placementrules/placement-my-policy",
            "uid": "2a52a098-b228-11e9-926f-eae85572ef1a",
          },
          "spec": Object {
            "clusterSelector": Object {
              "matchLabels": Object {
                "cloud": "IBM",
              },
            },
            "resourceHint": Object {},
            "resourceSelector": Object {},
          },
          "status": Object {
            "decisions": Array [
              Object {
                "clusterName": "clusterhub",
                "clusterNamespace": "clusterhub",
              },
              Object {
                "clusterName": "cluster1",
                "clusterNamespace": "cluster1",
              },
            ],
          },
        },
        "resourceSelector": Object {},
        "status": Object {
          "decisions": Array [
            Object {
              "clusterName": "clusterhub",
              "clusterNamespace": "clusterhub",
            },
            Object {
              "clusterName": "cluster1",
              "clusterNamespace": "cluster1",
            },
          ],
        },
      },
      Object {
        "clusterLabels": Object {
          "matchLabels": Object {
            "cloud": "IBM",
          },
        },
        "clusterReplicas": null,
        "metadata": Object {
          "annotations": Object {
            "mcm.ibm.com/user-group": "c3lzdGVtOnNlcnZpY2VhY2NvdW50cyxzeXN0ZW06c2VydmljZWFjY291bnRzOmt1YmUtc3lzdGVtLHN5c3RlbTphdXRoZW50aWNhdGVk",
            "mcm.ibm.com/user-identity": "c3lzdGVtOnNlcnZpY2VhY2NvdW50Omt1YmUtc3lzdGVtOmRlZmF1bHQ=",
          },
          "creationTimestamp": "2019-07-29T17:45:39Z",
          "name": "placement-policy-ma",
          "namespace": "mcm",
          "selfLink": "/apis/apps.open-cluster-management.io/v1/namespaces/mcm/placementrules/placement-policy-ma",
        },
        "raw": Object {
          "apiVersion": "apps.open-cluster-management.io/v1",
          "kind": "PlacementRule",
          "metadata": Object {
            "annotations": Object {
              "mcm.ibm.com/user-group": "c3lzdGVtOnNlcnZpY2VhY2NvdW50cyxzeXN0ZW06c2VydmljZWFjY291bnRzOmt1YmUtc3lzdGVtLHN5c3RlbTphdXRoZW50aWNhdGVk",
              "mcm.ibm.com/user-identity": "c3lzdGVtOnNlcnZpY2VhY2NvdW50Omt1YmUtc3lzdGVtOmRlZmF1bHQ=",
            },
            "creationTimestamp": "2019-07-29T17:45:39Z",
            "name": "placement-policy-ma",
            "namespace": "mcm",
            "resourceVersion": "98",
            "selfLink": "/apis/apps.open-cluster-management.io/v1/namespaces/mcm/placementrules/placement-policy-ma",
            "uid": "ada3957f-b228-11e9-926f-eae85572ef1a",
          },
          "spec": Object {
            "clusterSelector": Object {
              "matchLabels": Object {
                "cloud": "IBM",
              },
            },
            "resourceHint": Object {},
            "resourceSelector": Object {},
          },
          "status": Object {
            "decisions": Array [
              Object {
                "clusterName": "clusterhub",
                "clusterNamespace": "clusterhub",
              },
              Object {
                "clusterName": "cluster1",
                "clusterNamespace": "cluster1",
              },
            ],
          },
        },
        "resourceSelector": Object {},
        "status": Object {
          "decisions": Array [
            Object {
              "clusterName": "clusterhub",
              "clusterNamespace": "clusterhub",
            },
            Object {
              "clusterName": "cluster1",
              "clusterNamespace": "cluster1",
            },
          ],
        },
      },
      Object {
        "clusterLabels": Object {
          "matchLabels": Object {
            "cloud": "IBM",
          },
        },
        "clusterReplicas": null,
        "metadata": Object {
          "annotations": Object {
            "mcm.ibm.com/user-group": "c3lzdGVtOnNlcnZpY2VhY2NvdW50cyxzeXN0ZW06c2VydmljZWFjY291bnRzOmt1YmUtc3lzdGVtLHN5c3RlbTphdXRoZW50aWNhdGVk",
            "mcm.ibm.com/user-identity": "c3lzdGVtOnNlcnZpY2VhY2NvdW50Omt1YmUtc3lzdGVtOmRlZmF1bHQ=",
          },
          "creationTimestamp": "2019-07-29T17:41:59Z",
          "name": "placement-my-policy",
          "namespace": "mcm",
          "selfLink": "/apis/apps.open-cluster-management.io/v1/namespaces/mcm/placementrules/placement-my-policy",
        },
        "raw": Object {
          "apiVersion": "apps.open-cluster-management.io/v1",
          "kind": "PlacementRule",
          "metadata": Object {
            "annotations": Object {
              "mcm.ibm.com/user-group": "c3lzdGVtOnNlcnZpY2VhY2NvdW50cyxzeXN0ZW06c2VydmljZWFjY291bnRzOmt1YmUtc3lzdGVtLHN5c3RlbTphdXRoZW50aWNhdGVk",
              "mcm.ibm.com/user-identity": "c3lzdGVtOnNlcnZpY2VhY2NvdW50Omt1YmUtc3lzdGVtOmRlZmF1bHQ=",
            },
            "creationTimestamp": "2019-07-29T17:41:59Z",
            "name": "placement-my-policy",
            "namespace": "mcm",
            "resourceVersion": "60",
            "selfLink": "/apis/apps.open-cluster-management.io/v1/namespaces/mcm/placementrules/placement-my-policy",
            "uid": "2a52a098-b228-11e9-926f-eae85572ef1a",
          },
          "spec": Object {
            "clusterSelector": Object {
              "matchLabels": Object {
                "cloud": "IBM",
              },
            },
            "resourceHint": Object {},
            "resourceSelector": Object {},
          },
          "status": Object {
            "decisions": Array [
              Object {
                "clusterName": "clusterhub",
                "clusterNamespace": "clusterhub",
              },
              Object {
                "clusterName": "cluster1",
                "clusterNamespace": "cluster1",
              },
            ],
          },
        },
        "resourceSelector": Object {},
        "status": Object {
          "decisions": Array [
            Object {
              "clusterName": "clusterhub",
              "clusterNamespace": "clusterhub",
            },
            Object {
              "clusterName": "cluster1",
              "clusterNamespace": "cluster1",
            },
          ],
        },
      },
      Object {
        "clusterLabels": Object {
          "matchLabels": Object {
            "cloud": "IBM",
          },
        },
        "clusterReplicas": null,
        "metadata": Object {
          "annotations": Object {
            "mcm.ibm.com/user-group": "c3lzdGVtOnNlcnZpY2VhY2NvdW50cyxzeXN0ZW06c2VydmljZWFjY291bnRzOmt1YmUtc3lzdGVtLHN5c3RlbTphdXRoZW50aWNhdGVk",
            "mcm.ibm.com/user-identity": "c3lzdGVtOnNlcnZpY2VhY2NvdW50Omt1YmUtc3lzdGVtOmRlZmF1bHQ=",
          },
          "creationTimestamp": "2019-07-29T17:45:39Z",
          "name": "placement-policy-ma",
          "namespace": "mcm",
          "selfLink": "/apis/apps.open-cluster-management.io/v1/namespaces/mcm/placementrules/placement-policy-ma",
        },
        "raw": Object {
          "apiVersion": "apps.open-cluster-management.io/v1",
          "kind": "PlacementRule",
          "metadata": Object {
            "annotations": Object {
              "mcm.ibm.com/user-group": "c3lzdGVtOnNlcnZpY2VhY2NvdW50cyxzeXN0ZW06c2VydmljZWFjY291bnRzOmt1YmUtc3lzdGVtLHN5c3RlbTphdXRoZW50aWNhdGVk",
              "mcm.ibm.com/user-identity": "c3lzdGVtOnNlcnZpY2VhY2NvdW50Omt1YmUtc3lzdGVtOmRlZmF1bHQ=",
            },
            "creationTimestamp": "2019-07-29T17:45:39Z",
            "name": "placement-policy-ma",
            "namespace": "mcm",
            "resourceVersion": "98",
            "selfLink": "/apis/apps.open-cluster-management.io/v1/namespaces/mcm/placementrules/placement-policy-ma",
            "uid": "ada3957f-b228-11e9-926f-eae85572ef1a",
          },
          "spec": Object {
            "clusterSelector": Object {
              "matchLabels": Object {
                "cloud": "IBM",
              },
            },
            "resourceHint": Object {},
            "resourceSelector": Object {},
          },
          "status": Object {
            "decisions": Array [
              Object {
                "clusterName": "clusterhub",
                "clusterNamespace": "clusterhub",
              },
              Object {
                "clusterName": "cluster1",
                "clusterNamespace": "cluster1",
              },
            ],
          },
        },
        "resourceSelector": Object {},
        "status": Object {
          "decisions": Array [
            Object {
              "clusterName": "clusterhub",
              "clusterNamespace": "clusterhub",
            },
            Object {
              "clusterName": "cluster1",
              "clusterNamespace": "cluster1",
            },
          ],
        },
      },
    ],
  },
}
`;
